### ISSUE TITLE
Possible Bug: Output XLSX will have too many rows

### ISSUE THREAD

**Issue by automationjunky** [kindly/flatterer#37]
I think I might have encountered a bug when generating XLSX:

```
[2023-01-06T06:15:42.741Z ERROR] Output XLSX will have too many rows 351881 in sheet `testoutput`, maximum allowed 1048576
```

The way I read that is that I'm under the max allowed, however, it's bombing out that I'll be over the threshold.  My json file has 351,881 objects and the entire output of the flatterer run is:

```
❯ flatterer testoutput.json output --xlsx
[2023-01-06T06:15:38.889Z INFO ] Starting processing input.
[2023-01-06T06:15:42.330Z INFO ] Finished processing 351881 value(s)
[2023-01-06T06:15:42.330Z INFO ] Analyzing input data
[2023-01-06T06:15:42.335Z INFO ] Writing final CSV files
[2023-01-06T06:15:42.342Z INFO ]     Writing 351881 row(s) to testoutput.csv
[2023-01-06T06:15:42.722Z INFO ] Writing final XLSX file
[2023-01-06T06:15:42.741Z ERROR] Output XLSX will have too many rows 351881 in sheet `testoutput`, maximum allowed 1048576
Traceback (most recent call last):
  File "/Users/x/.virtualenvs/flatterer/bin/flatterer", line 8, in <module>
    sys.exit(cli())
  File "/Users/x/.virtualenvs/flatterer/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/x/.virtualenvs/flatterer/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/x/.virtualenvs/flatterer/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/x/.virtualenvs/flatterer/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/x/.virtualenvs/flatterer/lib/python3.10/site-packages/flatterer/__init__.py", line 287, in cli
    flatten(input_file,
  File "/Users/x/.virtualenvs/flatterer/lib/python3.10/site-packages/flatterer/__init__.py", line 132, in flatten
    flatten_rs(input, output_dir, csv, xlsx, sqlite, parquet,
RuntimeError: Output XLSX will have too many rows 351881 in sheet `testoutput`, maximum allowed 1048576
```

**Comment by kindly** [comment]
Can you confirm you are on the latest version?

This should have been fixed in 0.16.2:

https://flatterer.opendata.coop/changelog.html#id4

**Comment by kindly** [comment]
@automationjunky I have found the issue.  It is working in multithreaded mode `--threads=2` but not in normal mode.  The fix will be soon.

**Comment by kindly** [comment]
@automationjunky this is fixed in 0.17.1
